lib/spack/llnl/util/filesystem.py

```diff
@@ -159,8 +163,36 @@ def groupid_to_group(x):
             shutil.copy(filename, backup_filename)
 
         try:
-            for line in fileinput.input(filename, inplace=True):
-                print(re.sub(regex, repl, line.rstrip('\n')))
+            extra_kwargs = {}
+            if sys.version_info > (3, 0):
+                extra_kwargs = {'errors': 'surrogateescape'}
```

Comment:
@lee218llnl You can try to do something like the above also in #13479 to avoid the error with Python 2.6: 
```console
extra_kwargs = {}
if sys.version_info > (3, 0):
￼    extra_kwargs = {'encoding': 'utf-8', 'errors': 'surrogateescape'}
```